modules/csg/csg_shape.cpp

```diff
@@ -1831,44 +1837,50 @@ CSGBrush *CSGPolygon::_build_brush() {
 	smooth.resize(face_count);
 	materials.resize(face_count);
 	invert.resize(face_count);
+	int faces_removed = 0;
 
-	PoolVector<Vector3>::Write facesw = faces.write();
-	PoolVector<Vector2>::Write uvsw = uvs.write();
-	PoolVector<bool>::Write smoothw = smooth.write();
-	PoolVector<Ref<Material>>::Write materialsw = materials.write();
-	PoolVector<bool>::Write invertw = invert.write();
-
-	int face = 0;
-	Transform base_xform;
-	Transform current_xform;
-	Transform previous_xform;
-	double u_step = 1.0 / extrusions;
-	double v_step = 1.0 / shape_sides;
-	double spin_step = Math::deg2rad(spin_degrees / spin_sides);
-	double extrusion_step = 1.0 / extrusions;
-	if (mode == MODE_PATH) {
-		if (path_joined) {
-			extrusion_step = 1.0 / (extrusions - 1);
-		}
-		extrusion_step *= curve->get_baked_length();
-	}
+	{
+		PoolVector<Vector3>::Write facesw = faces.write();
+		PoolVector<Vector2>::Write uvsw = uvs.write();
+		PoolVector<bool>::Write smoothw = smooth.write();
+		PoolVector<Ref<Material>>::Write materialsw = materials.write();
+		PoolVector<bool>::Write invertw = invert.write();
 
-	if (mode == MODE_PATH) {
-		if (!path_local) {
-			base_xform = path->get_global_transform();
+		int face = 0;
+		Transform base_xform;
+		Transform current_xform;
+		Transform previous_xform;
+		Transform previous_previous_xform;
+		double u_step = 1.0 / extrusions;
+		if (uv_distance > 0.0) {
+			u_step *= curve_length / uv_distance;
+		}
+		double v_step = 1.0 / shape_sides;
+		double spin_step = Math::deg2rad(spin_degrees / spin_sides);
+		double extrusion_step = 1.0 / extrusions;
+		if (mode == MODE_PATH) {
+			if (path_joined) {
+				extrusion_step = 1.0 / (extrusions - 1);
+			}
+			extrusion_step *= curve_length;
 		}
 
-		Vector3 current_point = curve->interpolate_baked(0);
-		Vector3 next_point = curve->interpolate_baked(extrusion_step);
-		Vector3 current_up = Vector3(0, 1, 0);
-		Vector3 direction = next_point - current_point;
+		if (mode == MODE_PATH) {
+			if (!path_local) {
+				base_xform = path->get_global_transform();
+			}
 
-		if (path_joined) {
-			Vector3 last_point = curve->interpolate_baked(curve->get_baked_length());
-			direction = next_point - last_point;
-		}
+			Vector3 current_point = curve->interpolate_baked(0);
+			Vector3 next_point = curve->interpolate_baked(extrusion_step);
+			Vector3 current_up = Vector3(0, 1, 0);
+			Vector3 direction = next_point - current_point;
 
-		switch (path_rotation) {
+			if (path_joined) {
+				Vector3 last_point = curve->interpolate_baked(curve->get_baked_length());
+				direction = next_point - last_point;
+			}
+
+			switch (path_rotation) {
 			case PATH_ROTATION_POLYGON:
```

Comment:
There are style issues with the lack of indentation for switch cases:
https://github.com/godotengine/godot/pull/52509/checks?check_run_id=3556178843

Reply:
Yeah, I fixed everything up for the master version: https://github.com/godotengine/godot/pull/52512  Figured it'd be best to get that approved first then come back to this one.  Not sure if that should be cherry-picked back into 3.x and this PR should be closed or if it's better to be independent, since the class names and stuff changed, likely making a merge problematic.

Reply:
Yeah finalizing the `master` PR before updating this one sounds good.